### PR TITLE
Updated certain variable names

### PR DESCRIPTION
**Pull request by TheBoringEdward**
Many of the variables used by the opponents are either misleading or outdated, which has been rectified within this branch .

Due to this branch originally being intended to implement a sort of jump-pad, said jump-pad can be found as a new opponent-type, although, other than having a different texture and name, being functionally identical to the already existing barriers.

@GemueseHasser , please verify first that this pull will not bugger something up within the master branch.

### CODE REVIEW

File: src/de/informatik/game/object/map/opponent/Barrier.java [hunk 1]
```diff
@@ -53,40 +53,28 @@ public void drawOpponent(final Graphics2D g) {
     }
 
     @Override
-    public void playerMoveLeftEvent(final int playerPosition, final boolean isBackgroundMovable) {
-        if (!isBackgroundMovable) return;
+    public void playerMoveLeftEvent(final int playerPosition, final boolean isBackgroundMoving) {
+        if (!isBackgroundMoving) return;
```
Comment (GemueseHasser) [inline src/de/informatik/game/object/map/opponent/Barrier.java]: we should not rename the isBackgroundMovable variable to isBackgroundMoving as this name would be incorrect.

File: src/de/informatik/game/object/map/opponent/Barrier.java [hunk 2]
```diff
@@ -53,40 +53,28 @@ public void drawOpponent(final Graphics2D g) {
...
-
-        // JumpAndRun.GAME_INSTANCE.getGameHandler().getMap().getLastMiddleBackgroundX() not updating properly
-        // always jumps back to 0 at -615, when wrapping background, due to the bg's x-coordinate resetting
     }
 
     @Override
-    public void playerMoveRightEvent(final int playerPosition, final boolean isBackgroundMovable) {
-        if (!isBackgroundMovable) return;
+    public void playerMoveRightEvent(final int playerPosition, final boolean isBackgroundMoving) {
+        if (!isBackgroundMoving) return;
```
Comment (GemueseHasser) [inline src/de/informatik/game/object/map/opponent/Barrier.java]: we should not rename the isBackgroundMovable variable to isBackgroundMoving as this name would be incorrect.

File: src/de/informatik/game/constant/OpponentType.java [hunk 3]
```diff
@@ -10,7 +11,8 @@ public enum OpponentType {
 
     //<editor-fold desc="VALUES">
     /** Der Typ einer einfachen Barriere. */
-    BARRIER(Barrier.class);
+    BARRIER(Barrier.class),
+    JUMPPAD(Jumppad.class);
```
Comment (GemueseHasser) [inline src/de/informatik/game/constant/OpponentType.java:16]: we should also write documentation for this new enum value.

File: src/de/informatik/game/constant/ImageType.java [hunk 4]
```diff
@@ -25,7 +25,8 @@ public enum ImageType {
     /** Die vierte Animation des Spielers. */
     UNC_4("player/unc4.png"),
     /** Eine einfache Barriere. */
-    BARRIER("barrier.png");
+    BARRIER("barrier.png"),
+    JUMPPAD("jumppad.png");
```
Comment (GemueseHasser) [inline src/de/informatik/game/constant/ImageType.java:30]: we should also write documentation for this new enum value.

File: src/de/informatik/game/object/map/opponent/Jumppad.java
Comment (GemueseHasser) [inline src/de/informatik/game/object/map/opponent/Jumppad.java:1]: It basically belongs to Javadoc about the class, all variables and all methods that are not overridden. In addition, all constant variables and all local variables should be ordered and enclosed in an editor fold. Two paragraphs should be inserted between each block consisting of variables. Likewise, all implemented methods should be enclosed in an editor fold. Each class should end with a blank line.

**Review by GemueseHasser — CHANGES_REQUESTED** [review]

File: src/de/informatik/game/object/map/opponent/Jumppad.java
Comment (TheBoringEdward) [reply src/de/informatik/game/object/map/opponent/Jumppad.java:1]: I have no clue how to fix that one.

File: src/de/informatik/game/object/map/opponent/Barrier.java (hunk 1 above)
Comment (TheBoringEdward) [reply src/de/informatik/game/object/map/opponent/Barrier.java]: Reverted

File: src/de/informatik/game/object/map/opponent/Barrier.java (hunk 2 above)
Comment (TheBoringEdward) [reply src/de/informatik/game/object/map/opponent/Barrier.java]: Reverted

File: src/de/informatik/game/constant/OpponentType.java (hunk 3 above)
Comment (TheBoringEdward) [reply src/de/informatik/game/constant/OpponentType.java:16]: Added

File: src/de/informatik/game/constant/ImageType.java (hunk 4 above)
Comment (TheBoringEdward) [reply src/de/informatik/game/constant/ImageType.java:30]: Added

File: src/de/informatik/game/object/map/opponent/Jumppad.java
Comment (TheBoringEdward) [reply src/de/informatik/game/object/map/opponent/Jumppad.java:1]: I think I resolved it now....

Comment (TheBoringEdward) [reply src/de/informatik/game/object/map/opponent/Jumppad.java:1]: Now perhaps..?

**Review by GemueseHasser — APPROVED** [review]

**Pull request merged**